openvino_tensorflow/deassign_clusters.cc

```diff
@@ -193,24 +292,145 @@ Status DeassignClusters(Graph* graph) {
+        ss_input_to_retval << src->name() << ":" << edge->src_output();
       }
+
+      // If the destination node lies within a cluster
+      // src here is one of the input nodes of the cluster
+      if (dst_clustered) {
+        std::stringstream ss;
+        ss << "ngraph_input_" << cluster_input_map[dst_cluster_idx].size();
+        std::string new_input_name = ss.str();
+
+        input_rename_map[std::make_tuple(dst_cluster_idx, src->name(),
+                                         edge->src_output())] = new_input_name;
+        string input_prov_tag = src->name();
+
+        if (cluster_graph_map.find(dst_cluster_idx) == cluster_graph_map.end())
+          cluster_graph_map[dst_cluster_idx] =
+              *NGraphClusterManager::GetClusterGraph(dst_cluster_idx);
+
+        auto new_input_node_def = cluster_graph_map[dst_cluster_idx].add_node();
+
+        new_input_node_def->set_name(new_input_name);
+        new_input_node_def->set_op("_Arg");
+
+        SetAttrValue(dt, &((*(new_input_node_def->mutable_attr()))["T"]));
+        SetAttrValue(arg_index_count[dst_cluster_idx],
+                     &((*(new_input_node_def->mutable_attr()))["index"]));
+        SetAttrValue(input_prov_tag,
+                     &((*(new_input_node_def->mutable_attr()))["_prov_tag"]));
+
+        arg_index_count[dst_cluster_idx]++;
+        cluster_input_map[dst_cluster_idx].push_back(
+            std::make_tuple(src->id(), edge->src_output(), dt));
+      }
+    }
+
+    for (auto& c : cluster_graph_map) {
+      int cluster_idx = c.first;
+      GraphDef graph_def = c.second;
+
+      PopulateClusterGraphDef(cluster_idx, graph_def, graph, input_rename_map);
+
+      GraphConstructorOptions opts;
+      opts.allow_internal_ops = true;
+      opts.expect_device_spec = false;
+      FunctionLibraryDefinition flib(OpRegistry::Global(), graph_def.library());
+      Graph g(flib);
+      TF_RETURN_IF_ERROR(ConvertGraphDefToGraph(opts, graph_def, &g));
+
+      int num_nodes = 0;
+      std::pair<int, int64_t> pair_idx_cost;
+      pair_idx_cost.first = cluster_idx;
+      pair_idx_cost.second = 0;
+      for (auto n : g.nodes()) {
+        int64_t node_cost = 0;
+        if (GetNodeAttr(n->attrs(), "cost", &node_cost) != Status::OK())
+          continue;
+        pair_idx_cost.second += node_cost;
+        num_nodes++;
+      }
+      pair_idx_cost.second /= 1e6;  // ns to ms
+      cluster_cost_map_in_ms.push_back(pair_idx_cost);
+
+      OVTF_VLOG(1) << "Cluster with ID " << cluster_idx
+                   << ", num_nodes: " << num_nodes << ", costs "
+                   << pair_idx_cost.second << " ms";
+    }
+
+    // sort clusters based on cost
+    std::sort(
+        cluster_cost_map_in_ms.begin(), cluster_cost_map_in_ms.end(),
+        [](auto& left, auto& right) { return left.second > right.second; });
+
+    // Keep only Top K clusters, based on the value of K set by the following
+    // environment variable
+    int top_k_clusters = 3;
```

Comment:
This is empirically validated on a good set of models and available for experimental use only for now. however it may need more experiments to finalize this logic for GA release of these feature.